15_restaurant.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# Implementation for RubyMonk Primer Problem 15 Orders and costs
+# Problem Statement:
+# A restaurant has incoming orders for which you need to compute the costs based on the menu.
+# You might get multiple orders at times.
+# Examples:
+# If the order is {:rice => 1, :noodles => 1}, and the menu is {:rice => 3, :noodles => 2},
+# then the cost is 5.
+# If the orders are {:rice => 1, :noodles => 1} and {:rice => 2, :noodles => 2} and the menu
+# is {:rice => 3, :noodles => 2}, then the cost is 15.
+class Restaurant
+  def initialize(menu)
+    @menu = menu
+  end
+
+  def cost(*orders)
+    # your code here
+    # total cost = sum of cost of each order
+    orders.reduce(0) do |total_cost, order|
+      total_cost + order.keys.reduce(0) { |order_cost, item| order_cost + @menu[item] * order[item] }
```

Comment:
This seems like a great candidate for a separate method!  But tell me -- does this read clearly to you (it reads nicely to me): `orders.map { |order| order_cost(order) }.reduce(&:+)`?

If so, just split out the `order_cost` part of the equation.

Reply:
Yes, I agree with all of this and what a great way to convince me to change my code.

Reply:
✅